3rdparty/libtim-vx/tim-vx.cmake

```diff
@@ -0,0 +1,58 @@
+#set(TIMVX_COMMIT_HASH "e001d53ddfe1df86838d2bc593d043358e8a4527")
+set(TIMVX_COMMIT_HASH "8051050cb865b04bf28a8febf7ed4e13626a9706")
+set(OCV_TIMVX_DIR "${OpenCV_BINARY_DIR}/3rdparty/libtim-vx")
+set(OCV_TIMVX_SOURCE_PATH "${OCV_TIMVX_DIR}/TIM-VX-${TIMVX_COMMIT_HASH}")
+
+if(EXISTS "${OCV_TIMVX_SOURCE_PATH}")
+    message(STATUS "Tim-VX exists already at ${OCV_TIMVX_SOURCE_PATH}")
+
+    set(TIMVX_FOUND ON)
+    set(BUILD_TIMVX ON)
+else()
+    set(OCV_TIMVX_FILENAME "${TIMVX_COMMIT_HASH}.zip")
+    #set(OCV_TIMVX_URL "https://github.com/VeriSilicon/TIM-VX/archive/")
+    set(OCV_TIMVX_URL "https://github.com/fengyuentau/TIM-VX/archive/")
+    #set(timvx_zip_md5sum 4e7bdd168d76e6937096132ca430791d)
+    set(timvx_zip_md5sum 4d31e42b98f18f03e7369f59b95c6cac)
+
+    ocv_download(FILENAME ${OCV_TIMVX_FILENAME}
+                 HASH ${timvx_zip_md5sum}
+                 URL "${OCV_TIMVX_URL}"
+                 DESTINATION_DIR "${OCV_TIMVX_DIR}"
+                 ID TIMVX
+                 STATUS res
+                 UNPACK RELATIVE_URL)
+
+    if(NOT res)
+        set(TIMVX_FOUND OFF)
+        message(STATUS "Failed to download Tim-VX source code from github. Turning off TIMVX_FOUND")
+    else()
+        set(TIMVX_FOUND ON)
+        set(BUILD_TIMVX ON)
+        message(STATUS "TIMVX: Downloaded Tim-VX.")
+    endif()
+endif()
+
+if(BUILD_TIMVX)
+    set(HAVE_TIMVX 1)
+
+    ocv_warnings_disable(CMAKE_C_FLAGS -Wunused-parameter -Wstrict-prototypes -Wundef -Wsign-compare -Wmissing-prototypes -Wmissing-declarations -Wstrict-aliasing -Wunused-but-set-variable -Wmaybe-uninitialized -Wshadow -Wsuggest-override)
+    ocv_warnings_disable(CMAKE_CXX_FLAGS -Wunused-parameter -Wstrict-prototypes -Wundef -Wsign-compare -Wunused-but-set-variable -Wshadow -Wsuggest-override -Wmissing-declarations)
+
+    set(TIMVX_SDK_DIR "${OCV_TIMVX_SOURCE_PATH}/prebuilt-sdk/x86_64_linux")
+
+    #set(BUILD_IN_OPENCV ON)
+    set(TimVX_INCLUDE_DIR "${OCV_TIMVX_SOURCE_PATH}/include" CACHE INTERNAL "")
+    if(EXISTS "${OCV_TIMVX_SOURCE_PATH}/CMakeLists.txt")
+        #set(BUILD_SHARED_LIBS OFF CACHE "" INTERNAL FORCE)
+        add_subdirectory("${OCV_TIMVX_SOURCE_PATH}" "${OCV_TIMVX_DIR}/build")
+        #set(BUILD_SHARED_LIBS ON CACHE "" INTERNAL FORCE)
+    else()
+        message(WARNING "Tim-VX: Missing 'CMakeLists.txt' in the source code: ${OCV_TIMVX_SOURCE_PATH}")
+    endif()
+    #if(BUILD_IN_OPENCV)
+    #    ocv_install_target(tim_internal EXPORT OpenCVModules ARCHIVE DESTINATION ${OPENCV_3P_LIB_INSTALL_PATH} COMPONENT dev)
+    #    ocv_install_target(tim-vx EXPORT OpenCVModules ARCHIVE DESTINATION ${OPENCV_3P_LIB_INSTALL_PATH} COMPONENT dev)
```

Comment:
`ocv_install_target` is now moved to my fork of TIM-VX. Maybe we can create a branch in the official TIM-VX repo just like `tengine-opencv` in Tengine.